samtranslator/model/role_utils/role_constructor.py

```diff
@@ -57,7 +57,7 @@ def _get_managed_policy_arn(
             if arn:
                 return arn
 
-    return name
+    return None
```

Comment:
what if it is an intrinsic? We should return as it is

Reply:
`name` is guaranteed a string:

https://github.com/aws/serverless-application-model/blob/b154e380d33cb0c28c9d015c11f5a5e929d6d99f/samtranslator/model/role_utils/role_constructor.py#L142-L148

Reply:
nvm we have `if isinstance(policy_arn, str):` in the caller.